src/backend/distributed/executor/multi_utility.c

```diff
@@ -280,10 +326,18 @@ multi_ProcessUtility(Node *parsetree,
 	{
 		if (IsA(parsetree, IndexStmt))
 		{
-			/* copy parse tree since we might scribble on it to fix the schema name */
-			parsetree = copyObject(parsetree);
+			/*
+			 * copy parse tree since we might scribble on it to fix the schema name
+			 * FIXME: PostgreSQL 10 changes caching behavior and breaks this approach,
+			 *        so for that version we do assign back the modified tree.
+			 */
```

Comment:
See the commit message for b2d5e3c and Marco's email about the topic.

At any rate, I just noticed this comment has a typo.